flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -101,6 +101,24 @@ function revert_flink_dir() {
     CURL_SSL_ARGS=""
 }
 
+function use_zookeeper_34() {
+  if [ -e "${FLINK_DIR}"/lib/flink-shaded-zookeeper-3.5* ]; then
+    # contents of 'opt' must not be changed since it is not backed up in common.sh#backup_flink_dir
+    # it is fine to delete jars from 'lib' since it is backed up and will be restored after the test
+    rm "${FLINK_DIR}"/lib/flink-shaded-zookeeper-3.5*
+    cp "${FLINK_DIR}"/opt/flink-shaded-zookeeper-3.4* "${FLINK_DIR}/lib"
```

Comment:
Where does `opt/zookeeper-3.4` come from? I think per default only `zookeeper-3.5` is in `opt`.

Reply:
technically speaking we don't need it and we could make the assumption in the test that 3.4 is _always_ in dist and 3.5 is _always_ in opt; but it felt wrong to not do _anything_ if 3.4 is specified as the ZK version.

With this it is at least possible to make 3.5 the default ZK version (which we may do in a later release) without requiring changes to the test.

Reply:
we could future-proof this a bit more by deleting `flink-shaded-zookeeper-3.*` from `lib` instead in both cases, and explicitly check that the 3.4/3.5 jar is present in opt before copying.